packages/rum/src/domain/record/record.ts

```diff
@@ -80,21 +78,23 @@ export function record(options: RecordOptions): RecordAPI {
       domEvents: [DOM_EVENT.CHANGE],
     })
     shadowDomCallBacks.set(shadowRoot, {
-      flush: () => shadowDomMutationController.flush(),
+      flush,
       stop: () => {
         stopMutationObserver()
         stopInputObserver()
       },
     })
   }
 
-  const documentMutationController = new MutationController()
   const takeFullSnapshot = (
     timestamp = timeStampNow(),
-    serializationContext = { status: SerializationContextStatus.INITIAL_FULL_SNAPSHOT, elementsScrollPositions }
+    serializationContext = { status: SerializationContextStatus.INITIAL_FULL_SNAPSHOT, elementsScrollPositions },
+    flushMutationsFromPreviousFs?: () => void
   ) => {
     shadowDomCallBacks.forEach(({ flush }) => flush())
-    documentMutationController.flush() // process any pending mutation before taking a full snapshot
+    if (flushMutationsFromPreviousFs) {
+      flushMutationsFromPreviousFs() // process any pending mutation before taking a full snapshot
+    }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Remove those lines from `takeFullSnapshot`.  Flush the mutations before calling `takeFullSnapshot()`, when needed. This should simplify things a bit